### PR TITLE
Trading View Demo

### PR DESCRIPTION
**Pull request by davidsoft318**
Hi, @chillvybz 

Please check trading view demo
Thank you.

### CODE REVIEW

File: public/config/main.config.json [hunk 1]
```diff
@@ -53,16 +58,21 @@
     },
     "defaultPair": {
       "to": "0xc2132D05D31c914a87C6611C10748AEb04B58e8F",
-      "from": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
+      "from": "0x0000000000000000000000000000000000001010"
```
Comment (chillvybz) [inline public/config/main.config.json:61]: Any reason you had to change this address?

Comment (davidkoller318) [reply public/config/main.config.json:61]: @chillvybz 
MATIC contract address is 0x0000000000000000000000000000000000001010 in Polygonscan.
When fetch the price info from coingecko api, we should send the exact contract address as a param. 

This address should be 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE ?

Comment (davidkoller318) [reply public/config/main.config.json:61]: Do we have any problems?

File: src/client/js/components/partials/swap/TradingViewChart.jsx [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+    line: [
+      {name: "1D", from: 'Past 1 day'},
+      {name: "3D", from: 'Past 3 days'},
+      {name: "1W", from: 'Past week'},
+      {name: "1M", from: 'Past month'},
+      {name: "1Y", from: 'Past year'}
+    ]
+  };
+
+  const wrapTokens = {
```
Comment (chillvybz) [inline src/client/js/components/partials/swap/TradingViewChart.jsx:28]: why are these hard-coded?

File: src/client/js/utils/tokenList.js [hunk 3]
```diff
@@ -22,8 +25,8 @@ window.TokenListManager = {
   },
 
   updateNetwork: function(network, connectStrategy) {
+    window.SELECTED_NETWORK = network.name;
```
Comment (chillvybz) [inline src/client/js/utils/tokenList.js:28]: you can remove this line. I think bad conflict here. I moved logic to `Storage.updateNetwork`

File: src/client/js/index.js [hunk 4]
```diff
@@ -46,6 +46,9 @@ var config  = await fetch(
 );
 window.NETWORK_CONFIGS = await config.json();
 
+// initialize TokenList
+window.COINGECKO_TOKEN_LIST = await(await fetch('/tokens/coingecko.list.json')).json();
```
Comment (chillvybz) [inline src/client/js/index.js:50]: How does the coingecko list diff from our list?

Comment (chillvybz) [reply src/client/js/index.js:50]: Also where is this file? I don't see it here

File: src/client/js/utils/tokenList.js [hunk 5]
```diff
@@ -76,6 +79,23 @@ window.TokenListManager = {
...
+    // update swap token configuration
+    const swap = {
+      from: this.findTokenById(network.defaultPair.from),
+      to: this.findTokenById(network.defaultPair.to),
+      network: network.name
+    }
+    this.updateSwapConfig(swap);
+  },
+
+  updateSwapConfig: function(swap) {
```
Comment (chillvybz) [inline src/client/js/utils/tokenList.js:91]: I understand why you introduced this "swap config".
But seems poorly introduced.

If I understand correctly, the goals are:
1) Updating the token in the SwapWidget should update the Pair in the TradingChart to display
2) Switching network, should update the SwapWidget and the TradingChart
3) Make any changes in TradingChat (if possible), should also switch SwapWidget pair
4) Future components include the GlobalSearchBar across the top, which should update both SwapWidget and TradingChart simultaneously.

Basically, the TradingChart and SwapWidget should display the same Pair.

I think this TokenList utility file could take on a larger role of managing:
- current network
- token list
- focused pair on the dashboard

**Review by chillvybz — CHANGES_REQUESTED** [review]

File: src/client/js/index.js (hunk 4 above)
Comment (davidkoller318) [reply src/client/js/index.js:50]: Sorry, I pushed it.

File: src/client/js/components/partials/swap/TradingViewChart.jsx (hunk 2 above)
Comment (davidkoller318) [reply src/client/js/components/partials/swap/TradingViewChart.jsx:28]: you mean wrapToken?

const wrapTokens = {
    "BNB": "0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c",
    "AVAX": "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7"
  }

As you know, when fetch the price info from Coingecko service, we should send the contract address as a param.
Coingecko api doesn't support BNB, AVAX contract address for fetching price. 
It support WBNB instead of BNB,  WAVAX instead of AVAX.

File: src/client/js/utils/tokenList.js (hunk 3 above)
Comment (davidkoller318) [reply src/client/js/utils/tokenList.js:28]: Okay. removed.

File: src/client/js/utils/tokenList.js (hunk 5 above)
Comment (davidkoller318) [reply src/client/js/utils/tokenList.js:91]: 1. Updating the token in the SwapWidget should update the Pair in the TradingChart to display. (Correct. it works like that)
2. Switching network, should update the SwapWidget and the TradingChart(Correct. it works like that)
3. Make any changes in TradingChat (if possible), should also switch SwapWidget pair
I referenced many popular dex site like  1inch, Paraswap, Uniswap etc. 
But  When make any changes in TradingChat ,  it doesn't switch swapWidget. 
For example, When token pair is BNB, USDC.   TradingChart displays 4 price charts.  BNB/USDC,  USDC/BNB, BNB, USDC. 
I hope you to run the trading view demo on local.
Thank you.

4. Future components include the GlobalSearchBar across the top, which should update both SwapWidget and TradingChart simultaneously. (Correct. it works like that)

Comment (davidkoller318) [reply src/client/js/utils/tokenList.js:91]: Let's discuss about  this in detail.

**Comment by chillvybz** [comment]
Issues:

1) Please fix this warning:
image

2) Spacing between the columns is off. Really big in the middle between SwapWidget and TradingChart. Also there is a gap on the right side of the SwapWidget that doesn't align with the NavBar
image

3) Somthing happeneded to the SWAP func. Pricing not working:
image

**Comment by davidkoller318** [comment]
Okay. got it.

**Comment by davidkoller318** [comment]
Hi, @chillvybz 
I fixed the issues.

**Comment by chillvybz** [comment]
Still looking for the following updates:
-  loading view when fetching the price on trading view.
-  if there is no price data, Display “We are unable to fetch the data for the selected pair”

**Comment by davidkoller318** [comment]
> Still looking for the following updates:
> 
> * loading view when fetching the price on trading view.
> * if there is no price data, Display “We are unable to fetch the data for the selected pair”

Hi, @chillvybz 
I updated above task.
Please review it. 
Thank you

**Comment by chillvybz** [comment]
image

It doesn't seem to scale well when the browser window re-sized

File: src/client/js/index.js (hunk 4 above)
Comment (chillvybz) [reply src/client/js/index.js:50]: Can you put this instructions as a code comment?
It is good documentation.

**Comment by chillvybz** [comment]
This looks good outside of the browser resize issue.
The only other thing is related to how we should handle the global state for the current PAIR we are looking at. This comment here:
 https://github.com/polkaswitch/polkaswitch-ui/pull/41#discussion_r699408490

We can talk about this in a meeting.

**Comment by chillvybz** [comment]
Closing. merged into #52
